components/omega/src/ocn/CustomTendencyTerms.cpp

```diff
@@ -0,0 +1,120 @@
+int ManufacturedSolution::init() {
+   int Err;
+
+   // Get ManufacturedSolConfig group
+   Config *OmegaConfig = Config::getOmegaConfig();
+   Config ManufacturedSolConfig("ManufacturedSolution");
+   Err = OmegaConfig->get(ManufacturedSolConfig);
+   if (Err != 0) {
+      LOG_CRITICAL("ManufacturedSolution:: ManufacturedSolution group "
+                   "not found in Config");
+      return Err;
+   }
+
+   // Get TendConfig group
+   Config TendConfig("Tendencies");
+   Err = OmegaConfig->get(TendConfig);
+   if (Err != 0) {
+      LOG_CRITICAL("ManufacturedSolution:: Tendencies group "
+                   "not found in Config");
+      return Err;
+   }
+
+   // Get manufactured solution parameters from Config
+   R8 WavelengthX;
+   R8 WavelengthY;
+   R8 Amplitude;
+
+   Err = ManufacturedSolConfig.get("WavelengthX", WavelengthX);
+   if (Err != 0) {
+      LOG_ERROR("ManufacturedSolution:: WavelengthX not found in "
+                "ManufacturedSolConfig");
+      return Err;
+   }
+
+   Err = ManufacturedSolConfig.get("WavelengthY", WavelengthY);
+   if (Err != 0) {
+      LOG_ERROR("ManufacturedSolution:: WavelengthY not found in "
+                "ManufacturedSolConfig");
+      return Err;
+   }
+
+   Err = ManufacturedSolConfig.get("Amplitude", Amplitude);
+   if (Err != 0) {
+      LOG_ERROR("ManufacturedSolution:: Amplitude not found in "
+                "ManufacturedSolConfig");
+      return Err;
+   }
+
+   // Get Tendendices parameters for del2 and del4 source terms
+   Err = TendConfig.get("VelDiffTendencyEnable",
+                        ManufacturedVelTend.VelDiffTendencyEnable);
+   Err += TendConfig.get("VelHyperDiffTendencyEnable",
+                         ManufacturedVelTend.VelHyperDiffTendencyEnable);
+   Err += TendConfig.get("ViscDel2", ManufacturedVelTend.ViscDel2);
+   Err += TendConfig.get("ViscDel4", ManufacturedVelTend.ViscDel4);
+
+   if (Err != 0) {
+      LOG_ERROR("ManufacturedSolution::Error reading Tendencies config");
+      return Err;
+   }
+
+   // Get the reference time to compute the model elapsed time
+   /// Get model clock from time stepper
+   TimeStepper *DefStepper             = TimeStepper::getDefault();
+   Clock *ModelClock                   = DefStepper->getClock();
+   ManufacturedThickTend.ReferenceTime = ModelClock->getCurrentTime();
+   ManufacturedVelTend.ReferenceTime   = ManufacturedThickTend.ReferenceTime;
+
+   // Get BottomDepth for the resting thickness
+   /// This test case assumes that the restingThickness is horizontally uniform
+   /// and that only one vertical level is used so only one set of indices is
+   /// used here.
+   HorzMesh *DefHorzMesh = HorzMesh::getDefault();
+   R8 H0                 = DefHorzMesh->BottomDepthH(0);
+
+   // Define and compute common constants
+   R8 Grav    = 9.80665_Real;                          // Gravity acceleration
+   R8 Pii     = 3.141592653589793_Real;                // Pi
```

Comment:
This is outside the scope of this PR, but should we start thinking about creating a shared constants file ? @mark-petersen @xylar Thoughts ?

Reply:
These need (!!!) to come from:
https://github.com/E3SM-Project/E3SM/blob/09b9b7e21e3d381eafa962e178a5951580e2fc74/share/util/shr_const_mod.F90

How is Scream handling this?

Reply:
Not sure that they do:

https://github.com/E3SM-Project/E3SM/blob/09b9b7e21e3d381eafa962e178a5951580e2fc74/components/eamxx/src/physics/share/physics_constants.hpp#L46

https://github.com/E3SM-Project/E3SM/blob/09b9b7e21e3d381eafa962e178a5951580e2fc74/share/util/shr_const_mod.F90#L16

Reply:
Agreed. Common constants need to come from a unified file, and preferably the centralized one that @xylar pointed to above. We struggled with multiple constants files in MPAS within E3SM.